ext/pgsql/config.m4

```diff
@@ -1,76 +1,38 @@
 PHP_ARG_WITH([pgsql],
   [for PostgreSQL support],
   [AS_HELP_STRING([[--with-pgsql[=DIR]]],
-    [Include PostgreSQL support. DIR is the PostgreSQL base install directory or
-    the path to pg_config])])
+    [Include PostgreSQL support. Optional DIR is the PostgreSQL base install
+    directory or the path to pg_config. When using pkg-config, the PGSQL_CFLAGS
+    and PGSQL_LIBS environment variables can be used instead of the DIR argument
+    to customize the libpq paths.])])
 
 if test "$PHP_PGSQL" != "no"; then
-  PHP_EXPAND_PATH($PGSQL_INCLUDE, PGSQL_INCLUDE)
+  PHP_SETUP_PGSQL([PGSQL_SHARED_LIBADD],,, [$PHP_PGSQL])
+  PHP_SUBST([PGSQL_SHARED_LIBADD])
 
-  dnl pg_config is still the default way to retrieve build options
-  dnl pkgconfig support was only introduced in 9.3
-
-  AC_MSG_CHECKING(for pg_config)
-  for i in $PHP_PGSQL $PHP_PGSQL/bin /usr/local/pgsql/bin /usr/local/bin /usr/bin ""; do
-	if test -x $i/pg_config; then
-      PG_CONFIG="$i/pg_config"
-      break;
-    fi
-  done
-
-  if test -n "$PG_CONFIG"; then
-    AC_MSG_RESULT([$PG_CONFIG])
-    PGSQL_INCLUDE=`$PG_CONFIG --includedir`
-    PGSQL_LIBDIR=`$PG_CONFIG --libdir`
-  else
-    AC_MSG_RESULT(not found)
-    if test "$PHP_PGSQL" = "yes"; then
-      PGSQL_SEARCH_PATHS="/usr /usr/local /usr/local/pgsql"
-    else
-      PGSQL_SEARCH_PATHS=$PHP_PGSQL
-    fi
-
-    for i in $PGSQL_SEARCH_PATHS; do
-      for j in include include/pgsql include/postgres include/postgresql ""; do
-        if test -r "$i/$j/libpq-fe.h"; then
-          PGSQL_INC_BASE=$i
-          PGSQL_INCLUDE=$i/$j
-        fi
-      done
-
-      for j in lib $PHP_LIBDIR/pgsql $PHP_LIBDIR/postgres $PHP_LIBDIR/postgresql ""; do
-        if test -f "$i/$j/libpq.so" || test -f "$i/$j/libpq.a"; then
-          PGSQL_LIBDIR=$i/$j
-        fi
-      done
-    done
-  fi
-
-  if test -z "$PGSQL_INCLUDE"; then
-    AC_MSG_ERROR(Cannot find libpq-fe.h. Please specify correct PostgreSQL installation path)
-  fi
-
-  if test -z "$PGSQL_LIBDIR"; then
-    AC_MSG_ERROR(Cannot find libpq.so. Please specify correct PostgreSQL installation path)
-  fi
+  AC_DEFINE(HAVE_PGSQL,1,[Whether to build PostgreSQL support or not])
 
-  if test -z "$PGSQL_INCLUDE" && test -z "$PGSQL_LIBDIR"; then
-    AC_MSG_ERROR([Unable to find libpq anywhere under $PGSQL_SEARCH_PATHS])
-  fi
+  PHP_CHECK_LIBRARY([pq], [lo_truncate64],
+    [AC_DEFINE([HAVE_PG_LO64], [1], [PostgreSQL 9.3 or later])],,
+    [$PGSQL_LIBS])
+  PHP_CHECK_LIBRARY([pq], [PQsetErrorContextVisibility],
+    [AC_DEFINE([HAVE_PG_CONTEXT_VISIBILITY], [1], [PostgreSQL 9.6 or later])],,
```

Comment:
If you aim for 10.0, perhaps checks and ifdefs for `HAVE_PG_LO64` and `HAVE_PG_CONTEXT_VISIBILITY` can be removed?

Reply:
Yes, this will be cleaned up. In this PR I'd only refactor the check into a separate macro. In the follow up PR, this would be bumped to 10.0 and these conditions removed. And also the PQlibVersion check would be replaced there with PQencryptPasswordConn (added in 10.0).